model/src/test/kotlin/com/testerum/model/resources/http/response/XmlBodyHttpResponseTest.kt

```diff
@@ -0,0 +1,188 @@
+            """xmlBody.bookstore.getChild("book")[0].getChild("title").getAttr("lang")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+
+        Assertions.assertEquals("en", expressionResult)
+    }
+
+    @Test
+    fun evaluateXPathToGetAttribute() {
+        val xml = """
+            <bookstore>
+              <book>
+                <title lang="en">Harry Potter</title>
+                <price>29.99</price>
+              </book>
+
+              <book>
+                <title lang="nl">Learning XML</title>
+                <price>39.95</price>
+              </book>
+            </bookstore>
+        """.trimIndent()
+
+        var expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.xpath("/*/book[1]/title/@lang")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("en", expressionResult)
+
+        expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.xpath("/*/book[2]/title/@lang")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("nl", expressionResult)
+    }
+
+    @Test
+    fun evaluateXPathToGetText() {
+        val xml = """
+            <bookstore>
+              <book>
+                <title lang="en">Harry Potter</title>
+                <price>29.99</price>
+              </book>
+
+              <book>
+                <title lang="nl">Learning XML</title>
+                <price>39.95</price>
+              </book>
+            </bookstore>
+        """.trimIndent()
+
+        var expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.xpath("/*/book[1]/title/text()")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("Harry Potter", expressionResult)
+
+        expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.xpath("/*/book[2]/title/text()")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("Learning XML", expressionResult)
+    }
+
+    @Test
+    fun evaluateXPathToGetMultipleTextNodes() {
+        val xml = """
+            <bookstore>
+              <book>
+                <title lang="en">Harry Potter</title>
+                <price>29.99</price>
+              </book>
+
+              <book>
+                super tare
+                <title lang="nl">Learning XML</title>
+                clar ca da
+                <price>39.95</price>
+                merge si asa
+              </book>
+            </bookstore>
+        """.trimIndent()
+
+        val expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.xpath("/*/book[2]/text()")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertLinesMatch(listOf(
+            "super tare",
+            "clar ca da",
+            "merge si asa"
+        ), expressionResult as List<String>)
```

Comment:
it's a bit strange that the ``getText()`` method returns a ``String``, while the equivalent xpath returns a ``List<String>``